autoPyTorch/datasets/base_dataset.py

```diff
@@ -81,16 +149,76 @@ def create_val_split(self,
                     '`val_share` specified, but the Dataset was a given a pre-defined split at initialization already.')
             if val_share < 0 or val_share > 1:
                 raise ValueError(f"`val_share` must be between 0 and 1, got {val_share}.")
-            if holdout_val_type not in self.cross_validators:
+            if not isinstance(holdout_val_type, HoldoutValTypes):
                 raise NotImplementedError(f'The specified `holdout_val_type` "{holdout_val_type}" is not supported.')
             kwargs = {}
             if is_stratified(holdout_val_type):
                 # we need additional information about the data for stratification
                 kwargs["stratify"] = self.train_tensors[-1]
-            train, val = self.holdout_validators[holdout_val_type](val_share, self._get_indices(), **kwargs)
+            train, val = self.holdout_validators[holdout_val_type.name](val_share, self._get_indices(), **kwargs)
             return Subset(self, train), Subset(self, val)
         else:
             if self.val_tensors is None:
                 raise ValueError('Please specify `val_share` or initialize with a validation dataset.')
             val = BaseDataset(self.val_tensors)
             return self, val
+
+    def get_dataset_for_training(self, split: int) -> Tuple[Dataset, Dataset]:
+        """
+        The above split methods employ the Subset to internally subsample the whole dataset.
+
+        During training, we need access to one of those splits. This is a handy function
+        to provide training data to fit a pipeline
+
+        Args:
+            split (int): The desired subset of the dataset to split and use
+
+        Returns:
+            Dataset: the reduced dataset to be used for testing
+        """
+        if isinstance(self.resampling_strategy, HoldoutValTypes):
+            # Regardless of the split, there is a single dataset
+            val_share = DEFAULT_RESAMPLING_PARAMETERS[self.resampling_strategy].get(
+                'val_share', None)
+            if self.resampling_strategy_args is not None:
+                val_share = self.resampling_strategy_args.get('val_share', val_share)
+            return self.create_val_split(
+                holdout_val_type=self.resampling_strategy,
+                val_share=val_share,
+            )
+        elif isinstance(self.resampling_strategy, CrossValTypes):
+            num_splits = DEFAULT_RESAMPLING_PARAMETERS[self.resampling_strategy].get(
+                'num_splits', None),
+            if self.resampling_strategy_args is not None:
+                num_splits = self.resampling_strategy_args.get('num_splits', num_splits)
+            # Create the split if it was not created before
+            self.create_cross_val_splits(
```

Comment:
If you create the warning as mentioned above, this could have an `if self.splits is None:`

Reply:
Ah, ok cool... processing this comment serially... so I will do that!